sherpa/models/model.py

```diff
@@ -640,6 +642,12 @@ def calc(self, p, *args, **kwargs):
                               type(self.rhs).__name__, len(rhs)))
         return val
 
+    def regrid(self, *arrays, **kwargs):
+        result = self.regrid_1d(*arrays, **kwargs)
+        for part in self:
+            part.regrid(*arrays)
+        return result
+
```

Comment:
I have two issues here:

a) this will only work for 1D models

b) does the `part.regrid` actually do anything useful (it returns no value, and I don't think it changes anything within the object)

Reply:
I've just tried commenting out the 'for part in self' case and the tests still pass, so I really don't understand what it's doing.

Reply:
Why `def regrid()`  is defined twice:  line 588 and 645?

Reply:
@anetasie - these are different classes.